### DISCUSSION TITLE
Have a command to add a launch,.json for running apps

### DISCUSSION THREAD

**Discussion by grab-a-byte** [ionide/ionide-vscode-fsharp#1827] [Ideas]
For some reason VS Code does not pick up that the project I have is a .NET Core (or just .NET form 7.0) and as such does not allow me to add a "Net Core Debug Configuration". 
This means I am unable to run and debug using the F5 key as I do for other languages. It would be cool to have a command added that woudl just add a debug configuration for you so that you did not have to use the F# tab and click the debug button. 
Small issue but for people who prefer using keybaord shortcuts an not using their mouse it's a much nicer experience.

**Comment by Krzysztof-Cieslak** [comment]
Cc: @baronfel 

I remember that we've done something around generating debugging configs automatically, right?

**Comment by baronfel** [comment]
Yeah, but those require the coreclr debug type to be recognized (which happens when Omnisharp initializes) - which the OP's message make me think isn't happening correctly.

**Reply by grab-a-byte** [reply]
Is there anything I can do to see if this is the case to help with debugging this?

**Reply by baronfel** [reply]
I'm not sure - I'd start by checking the omnisharp repo and seeing if they have any instructions for debugging their extension startup to be honest.

**Comment by grab-a-byte** [comment]
So update on this, creating a new console app (dotnet new console --lang F#), fresh install of vscode When Ionide is eanbled and the C# extensions (for omnisharp) is also enabled, hitting F5 does not give me the dotnet core option
![image](https://user-images.githubusercontent.com/30392071/221425208-ebfc8440-2191-40f4-951c-0b8ef573422c.png)

If I disable Ionide, I get the option for dotnet 5+ and dotnet core 
![image](https://user-images.githubusercontent.com/30392071/221425240-999bd8b6-39e8-4f7a-bcb2-e4160db8d7f1.png)

Thinking this may be a bug somewhere? If so i'm happy to file an issue but thought i'd check here in case i've done something daft first.

**Reply by baronfel** [reply]
Oh, that's really interesting! Good find, so maybe somehow we're overriding omnisharp's delivered debug configurations?

**Reply by grab-a-byte** [reply]
Unusre however if ther is any update is it possible to be tagged in the fix, would be nice to learn and know how this stuff works xD

**Reply by baronfel** [reply]
I think this is our fault: https://github.com/ionide/ionide-vscode-fsharp/blob/1b5bf5f1182187f379570f4f0bc7fce48ff18b3f/release/package.json#L814-L818

I think we're overriding the Omnisharp `coreclr` type here by accident.

**Reply by baronfel** [reply]
Gimme a sec to dig into it, but I think we just need to define a different debug 'type', even if the configurations we generate for users use the 'coreclr' debug type under the hood.

**Reply by grab-a-byte** [reply]
Should I file an issue for this btw as it's seems to be a bug rather than a discussion?

**Reply by grab-a-byte** [reply]
@baronfel , this seem to still be an issue. Should I file a bug for this? I'm amazed more poeple havent come across this.

**Reply by baronfel** [reply]
Hi, sorry for the delay - please do log an issue for it.

**Comment by grab-a-byte** [comment]
I also don't know if this helps or hinders but I went back to version 6.0.0 (12 months ago) and it didnt seeem to change anything after reloading VS Code. Unsure if I missed somethign to make ionide rechange it's configs in vscode?
